### PR TITLE
【Hackathon 5th No.97】add paddle tanh_shrink op 

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### Details:

add paddle tanh_shrink op

### Tickets:

waitting for 
- https://github.com/openvinotoolkit/openvino/pull/20078

### Reference:
Docs: https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/tanhshrink_cn.html#tanhshrink

Support data type:
- float32
- float64 (openvino do not support float64)
- float16 (only support on gpu)

UnitTest: https://github.com/PaddlePaddle/Paddle/blob/29059a349810a7cb895d84b90f1b177fed952791/test/legacy_test/test_activation_op.py#L1094

Code: https://github.com/PaddlePaddle/Paddle/blob/release/2.5/python/paddle/nn/functional/activation.py#L1404

https://github.com/PaddlePaddle/Paddle/issues/57262

### Test

waiting for:
- https://github.com/openvinotoolkit/openvino/pull/20161

### CODE REVIEW

**Comment by OpenVINO-dev-contest** [comment]
@yuxu42 Could you help to have a review on this PR, thanks

**Comment by yuxu42** [comment]
@xczhai Could you please help review? Thanks!

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_tanh_shrink.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def main():
+    data = np.random.uniform(10, 20, [2, 3, 4]).astype(np.float32)
+    tanh_shrink("tanh_shrink_1", data)
+
+    data = np.random.uniform(-10, 20, [4, 3, 2]).astype(np.float32)
+    tanh_shrink("tanh_shrink_2", data)
+
+    # data = np.random.uniform(10, 20, [2, 3, 4]).astype(np.float16) // only on gpu
```
Comment (xczhai) [inline src/frontends/paddle/tests/test_models/gen_scripts/generate_tanh_shrink.py:48]: please remove the useless codes if unnecessary.

Comment (Asthestarsfalll) [reply src/frontends/paddle/tests/test_models/gen_scripts/generate_tanh_shrink.py]: done

**Review by xczhai — APPROVED** [review]
LGTM

**Comment by xczhai** [comment]
@Asthestarsfalll please rebase the PR. triger the CI again. Thanks.

**Comment by xczhai** [comment]
@ceciliapeng2011 Hi Cecilia, could you help merge the PR? Thanks.

**Review by ceciliapeng2011 — APPROVED** [review]

**Pull request merged**